### DISCUSSION TITLE
How to proceed with code conformance fixes upstream

### DISCUSSION THREAD

**Discussion by roncapat** [space-ros/space-ros#276] [Ideas]
Simple question, but I think we may establish a standard approach on this (unless there isn't one already and I missed it).

Take as an example: https://github.com/space-ros/space-ros/issues/7

How to proceed upstream? I propose by issuing PRs with a sort of Space ROS standard text like:

> ### SpaceROS Code Conformance
> This PR addresses a code conformance issue detected by [Space ROS](https://github.com/space-ros) code analysis tools.
>
> **Tool**: <name of the tool> eg. clang-tidy
> **Issue**: <issue brief> eg. default arguments on virtual methods
> **Full (SARIF?) output**:
> `....`

...and then from https://github.com/space-ros/space-ros/issues/7 add a post referencing the opened PR for tracking purposes?

Since there are many things to fix upstream, I think we need an easy way to track the PRs and from time to time step over all the still to-be-merged ones to check if further support/info is needed upstream or simply to kindly trigger the maintainers.

We may expand on this in sort of CONTRIBUTING.md file to be linked on top of each meta-issue like https://github.com/space-ros/space-ros/issues/7.

**Comment by EzraBrooks** [comment]
Action item from July 27 meeting: document standard processes with templates, etc. for code conformance PRs upstream

**Comment by roncapat** [comment]
Since I have not been able to join the meeting, is there some additional feedback / something on which you could expand a bit? how may I help?
